src/tcgwars/logic/impl/gen4/DPTrainerKit.groovy

```diff
@@ -0,0 +1,151 @@
+package tcgwars.logic.impl.gen4;
+import tcgwars.logic.impl.gen4.DiamondPearl;
+import tcgwars.logic.impl.gen4.MysteriousTreasures;
+
+
+import static tcgwars.logic.card.HP.*;
+import static tcgwars.logic.card.Type.*;
+import static tcgwars.logic.card.CardType.*;
+import static tcgwars.logic.groovy.TcgBuilders.*;
+import static tcgwars.logic.groovy.TcgStatics.*
+import static tcgwars.logic.card.Resistance.ResistanceType.*
+
+
+import tcgwars.logic.card.*
+import tcgwars.logic.util.*;
+
+/**
+ * @author axpendix@hotmail.com
+ * @author wertercatt@wertercatt.com
+ */
+public enum DPTrainerKit implements LogicCardInfo {
+
+  BUIZEL_MANAPHY_1 ("Buizel", "1", Rarity.COMMON, [BASIC, POKEMON, _WATER_]),
+  FLOATZEL_MANAPHY_2 ("Floatzel", "2", Rarity.RARE, [STAGE1, EVOLUTION, POKEMON, _WATER_]),
+  GOLDEEN_MANAPHY_3 ("Goldeen", "3", Rarity.COMMON, [BASIC, POKEMON, _WATER_]),
+  MANAPHY_MANAPHY_4 ("Manaphy", "4", Rarity.RARE, [BASIC, POKEMON, _WATER_]),
+  PIPLUP_MANAPHY_5 ("Piplup", "5", Rarity.COMMON, [BASIC, POKEMON, _WATER_]),
+  PRINPLUP_MANAPHY_6 ("Prinplup", "6", Rarity.UNCOMMON, [STAGE1, EVOLUTION, POKEMON, _WATER_),
+  SEAKING_MANAPHY_7 ("Seaking", "7", Rarity.UNCOMMON, [STAGE1, EVOLUTION, POKEMON, _WATER_]),
+  TOTODILE_MANAPHY_8 ("Totodile", "8", Rarity.COMMON, [BASIC, POKEMON, _WATER_]),
+  DUSK_BALL_MANAPHY_9 ("Dusk Ball", "9", Rarity.UNCOMMON, [TRAINER, ITEM]);
+  ENERGY_SEARCH_MANAPHY_10 ("Energy Search", "10", Rarity.COMMON, [TRAINER, ITEM]);
+  POTION_MANAPHY_11 ("Potion", "11", Rarity.COMMON, [TRAINER, ITEM]);
+  WATER_ENERGY_MANAPHY_12 ("Water Energy", "12", Rarity.COMMON, [BASIC_ENERGY, ENERGY]);
+  GEODUDE_LUCARIO_1 ("Geodude", "1", Rarity.COMMON, [BASIC, POKEMON, _FIGHTING_]),
+  GRAVELER_LUCARIO_2 ("Graveler", "2", Rarity.UNCOMMON, [STAGE1, EVOLUTION, POKEMON, _FIGHTING_]),
+  LUCARIO_LUCARIO_3 ("Lucario", "3", Rarity.RARE, [STAGE1, EVOLUTION, POKEMON, _FIGHTING_]),
+  MACHOKE_LUCARIO_4 ("Machoke", "4", Rarity.UNCOMMON, [STAGE1, EVOLUTION, POKEMON, _FIGHTING_]),
+  MACHOP_LUCARIO_5 ("Machop", "5", Rarity.COMMON, [BASIC, POKEMON, _FIGHTING_]),
+  RIOLU_LUCARIO_6 ("Riolu", "6", Rarity.UNCOMMON, [BASIC, POKEMON, _FIGHTING_]),
+  STARLY_LUCARIO_7 ("Starly", "7", Rarity.COMMON, [BASIC, POKEMON, _COLORLESS_]),
+  ENERGY_SEARCH_LUCARIO_8 ("Energy Search", "9", Rarity.COMMON, [TRAINER, ITEM]);
+  POTION_LUCARIO_9 ("Potion", "10", Rarity.COMMON, [TRAINER, ITEM]);
+  QUICK_BALL_LUCARIO_10 ("Quick Ball", "11", Rarity.UNCOMMON, [TRAINER, ITEM]);
+  FIGHTING_ENERGY_LUCARIO_11 ("Fighting Energy", "12", Rarity.COMMON, [BASIC_ENERGY, ENERGY]);
+
+
+  static Type C = COLORLESS, R = FIRE, F = FIGHTING, G = GRASS, W = WATER, P = PSYCHIC, L = LIGHTNING, M = METAL, D = DARKNESS, Y = FAIRY, N = DRAGON;
+
+  protected CardTypeSet cardTypes;
+  protected String name;
+  protected Rarity rarity;
+  protected String collectionLineNo;
+
+  WorldCollection(String name, String collectionLineNo, Rarity rarity, List<CardType> cardTypes) {
```

Comment:
This line doesn't seem to have been updated from where it was copied. In addition, the file, enum, and any other references should probably use the full name like we do for all the other sets without any acronyms, so all `DPTrainerKit`s should be `DiamondPearlTrainerKit`

```suggestion
  DiamondPearlTrainerKit(String name, String collectionLineNo, Rarity rarity, List<CardType> cardTypes) {
```

Reply:
Fixed